### PR TITLE
[azmedien:show] Add support for another form of playlists

### PR DESCRIPTION
**Pull request by goggle**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [x] New feature

---

### Description

I have discovered that there is another form of playlists on [azmedien] sites, e.g. [this one](http://www.telezueri.ch/219-topic-aera-trump-hat-offiziell-begonnen). I have added support for these types of playlists.

Note: 
 - The URL for the new playlist types contain either the word 'topic' or 'themen'. But it is not guaranteed that URLs with 'topic' or 'themen' will point to the new playlist type, e.g. [this one](http://www.telem1.ch/264-themen-weihnachtswahnsinn) contains 'themen', but points to a regular playlist, which is already supported by the information extractor.
 - For the extraction of the playlist entries, we cannot use `self.url_result(url_base + m.group('url'), ie=AZMedienIE.ie_key())` on line 161, because this new playlist type might contain links to regular playlists, e.g. on [this](http://www.telezueri.ch/219-topic-aera-trump-hat-offiziell-begonnen) playlist, we have a link to [another playlist](http://www.telezueri.ch/86-show-talktaeglich/14211-episode-der-maechtigste-mann-der-welt).
 - I have adjusted the title extraction, so that it works with the new type of playlists and still works with the old ones.

### CODE REVIEW

File: youtube_dl/extractor/azmedien.py [hunk 1]
```diff
@@ -136,10 +155,17 @@ def _real_extract(self, url):
                 for m in re.finditer(
                     r'<a[^>]+data-real=(["\'])(?P<url>http.+?)\1', webpage)]
 
+        if not entries:
+            url_base = remove_end(base_url(url), '/')
+            entries = [
+                self.url_result(url_base + m.group('url'))
```
Comment (dstftw) [inline youtube_dl/extractor/azmedien.py]: Use `urljoin`.

File: youtube_dl/extractor/azmedien.py [hunk 2]
```diff
@@ -136,10 +155,16 @@ def _real_extract(self, url):
                 for m in re.finditer(
                     r'<a[^>]+data-real=(["\'])(?P<url>http.+?)\1', webpage)]
 
+        if not entries:
+            entries = [
+                self.url_result(urljoin(base_url(url), m.group('url')))
```
Comment (dstftw) [inline youtube_dl/extractor/azmedien.py]: Remove `base_url`.

**Comment by goggle** [comment]
Is this ready to merge, so that the work on [azmedien] can be considered as "finished", or is there anything else to do?

**Comment by dstftw** [comment]
Squash commits.

**Comment by goggle** [comment]
Done.

**Comment by dstftw** [comment]
Cherry-picked in f13da8af289d7d9365e34ef705a53ac62aa3b570, thanks.

**Pull request closed without merging**